src/Bicep.Core/Rewriters/RewriterHelper.cs

```diff
@@ -13,7 +13,9 @@ public static class RewriterHelper
     {
         public static (BicepFile bicepFile, bool hasChanges) Rewrite(Compilation prevCompilation, BicepFile bicepFile, Func<SemanticModel, SyntaxRewriteVisitor> rewriteVisitorBuilder)
         {
-            var semanticModel = new SemanticModel(prevCompilation, bicepFile, prevCompilation.SourceFileGrouping.FileResolver, prevCompilation.GetEntrypointSemanticModel().LinterAnalyzer);
+            // Sometimes bicepFile does not exist in the previous compilation, in which case we fall back on the compilation's entry point model
+            var prevModel = prevCompilation.SourceFileGrouping.FileResultByUri.ContainsKey(bicepFile.FileUri) ? prevCompilation.GetSemanticModel(bicepFile) : prevCompilation.GetEntrypointSemanticModel();
```

Comment:
I think we should be able to avoid using `GetEntrypointSemanticModel` and instead use the correct semanticModel if we:
1. Pass `semanticModel` in as a function arg rather than constructing it on line 18.
1. Output a new semantic model as an additional output tuple property if the Bicep file has changed on line 27.
1. Chain the two together when doing rewrite passes on line 40.

Reply:
This was added when the approach you describe caused a few test failures in the language server. [The `InsertResourceHandler` creates a new `inmemory:///generated.bicep` file and then passing it to `RewriteHelper::RewriteMultiple` to fix casing/formatting and to remove any read-only properties](https://github.com/Azure/bicep/blob/89ed67462c8009e8e01ba67e58bba26c495e500b/src/Bicep.LangServer/Handlers/InsertResourceHandler.cs#L191-L196). This code path relies on the existing compilation to supply configuration and a file resolver, but the supplied BicepFile will not exist in that compilation's file grouping.

Since `prevCompilation.Configuration` before this PR is equivalent to `prevCompilation.GetEntrypointSemanticModel().Configuration`, this update preserves the existing behavior of `InsertResourceHandler`

Reply:
Makes sense. Feels like it would be nice to have the ability to easily rebuild a `Compilation` / `SourceFileGrouping` with an updated `BicepFile`, but seems like we don't have an easy way to do this today.